packages/wallet/ui/src/util/SuggestChain.js

```diff
@@ -45,51 +32,36 @@ export async function suggestChain(networkConfig, caption = undefined) {
     bip44: {
       coinType,
     },
-    bech32Config: {
-      bech32PrefixAccAddr: 'agoric',
-      bech32PrefixAccPub: 'agoricpub',
-      bech32PrefixValAddr: 'agoricvaloper',
-      bech32PrefixValPub: 'agoricvaloperpub',
-      bech32PrefixConsAddr: 'agoricvalcons',
-      bech32PrefixConsPub: 'agoricvalconspub',
-    },
+    bech32Config,
     currencies: [stakeCurrency, stableCurrency],
     feeCurrencies: [stableCurrency],
     features: ['stargate', 'ibc-transfer'],
   };
-  await window.keplr.experimentalSuggestChain(chainInfo);
-  await window.keplr.enable(chainId);
+};
 
-  const offlineSigner = window.getOfflineSigner(chainId);
-  const cosmJS = await SigningStargateClient.connectWithSigner(
-    rpc,
-    offlineSigner,
-  );
+export async function suggestChain(
+  networkConfig,
+  /** @type {string} */ caption = undefined,
+  io = {},
+) {
```

Comment:
if `caption` can be undefined then it's not `string`. let's type the function signature instead of inline.

```suggestion
/** @type {{
  * networkConfig: NetworkConfig
  * caption?: string
  * io?: Partial<IoPowers>
  * }}
  */
export async function suggestChain(networkConfig,, caption, io = {}) {
```

Reply:
Though second thought, why make `io` optional? Please make them required so this module does not depend anything ambient. Then you can remove `/* global globalThis */`.

Reply:
Why indeed. It looks like we addressed the only call site in this regard.